internal/admission/admission.go

```diff
@@ -21,13 +25,74 @@ const (
 
 type GatewayMutationHook struct {
 	istioClient istioversionedclient.Interface
+	nsLister    corev1listers.NamespaceLister
 	decoder     *admission.Decoder
+	externalDNS *ExternalDNSConfig
 }
 
-func NewGatewayMutationHook(client istioversionedclient.Interface) *GatewayMutationHook {
+//ExternalDNSConfig passes configuration to the external DNS mutation behavior
+type ExternalDNSConfig struct {
+	enabled  bool
+	target   string
+	selector Selector
+}
+
+//Selector is a key value pair for matching annotations
+type Selector struct {
+	key   string
+	value string
+}
+
+//SetEnabled set the endabled field to a bool value
+func (edc *ExternalDNSConfig) SetEnabled(enabled bool) {
+	if edc != nil {
+		edc.enabled = enabled
+	}
+}
+
+//SetTarget sets the target field to a string value
+func (edc *ExternalDNSConfig) SetTarget(target string) {
+	if edc != nil {
```

Comment:
does it hurt? It seems odd to expect the caller to recover() or panic and doing nothing is consistent with the behavior of an existing nil pointer.

Reply:
That is exactly what the caller should do.  This represents some kind of misconfiguration.  All the nil check is doing is hiding potential bugs and turning this into a no-op.   Unless there is a reason to check for nil and create new, or de-reference before performing an operation.

Reply:
Would you be happier if I return an error value on all the setters in this case? I don't want to use builders because that precludes SetSelector from returning an error on a malformed input so it would necessarily become SetSelectorOrDie.

Reply:
I don't think we need to use builders here.  I just don't see a case where `edc` would be nil.  Any case where its `nil` would be an implementation error that would surface in unit tests.